pebblo/app/service/discovery/discovery_service.py

```diff
@@ -0,0 +1,250 @@
+        retrievals_details = list()
+
+        if app_metadata:
+            retrievals_details = app_metadata.get("retrievals", [])
+
+        input_retrievals_details = self.data.get("retrievals", [])
+        retrievals_details.extend(input_retrievals_details)
+
+        return retrievals_details
+
+    @timeit
+    def process_request(self, data):
+        try:
+            self.db = SQLiteClient()
+            self.data = data
+            self.app_name = data.get("name")
+
+            logger.debug("Discovery API request started")
+
+            # create session
+            self.db.create_session()
+
+            chain_details = []
+            retrievals_details = []
+            app_type, AppClass = self._get_app_type_and_class()
+            if not AppClass:
+                message = "No load_id's or run_id's are present, Invalid Request"
+                return self.return_response(message=message, status_code=404)
+
+            # get or create app
+            ai_app_obj = get_or_create_app(
+                self.db, self.app_name, AppClass, self.data, app_type
+            )
+            if not ai_app_obj:
+                message = "Unable to get or create aiapp details"
+                return self.return_response(message=message, status_code=500)
+
+            ai_app = ai_app_obj.data
+            # Get instance details
+            instance_details = self._fetch_runtime_instance_details()
+
+            if app_type == ApplicationTypes.RETRIEVAL.value:
+                # its retrieval application
+
+                # Get chain details
+                chain_details = self._fetch_chain_details(ai_app)
+
+                # get retrieval details
+                retrievals_details = self._fetch_retrievals_details(ai_app)
+
+            # Create AiApp Model
+            ai_apps_data = self.create_app_obj(
+                ai_app,
+                instance_details=instance_details,
+                chain_details=chain_details,
+                retrievals_details=retrievals_details,
+                app_type=app_type,
+            )
+
+            status, message = self.db.update_data(
+                table_obj=ai_app_obj, data=ai_apps_data
+            )
+            if not status:
+                logger.error(f"Process request failed: {message}")
+                return self.return_response(message=message, status_code=500)
+
+        except Exception as err:
+            logger.error(f"Discovery api failed, Error: {err}")
+            # Getting error, We are rollback everything we did in this run.
+            self.db.session.rollback()
+            return self.return_response(
+                message=f"Discovery api failed, Error: {err}", status_code=500
+            )
+
+        else:
+            # Commit will only happen when everything went well.
+            message = "App Discover Request Processed Successfully"
+            logger.debug(message)
```

Comment:
We should add an idetifier in the log messsage
This should be done in all the log messages